### PR TITLE
ofx: Search up to two parent accounts while finding matches

### LINKED ISSUE #113: beancount_import matching broken for `securities_and_cash` ofx account?
I just tried beancount-import with an investment account for the first time, and I'm having trouble getting it to match any of my existing transactions.

After some fiddling, I llearned that if I take a transaction generated by `beancount-import` and manually delete the metadata, `beancount-import` won't realize it's the same transaction, and will try to generate it again.

(Some numbers below are replaced with `XXX` for privacy.)

As an example, with the below stripped-down beancount file and `run_beancount_import.py`, `beancount-import` generates the following new directives:

```
2020-12-30 * "SELLSTOCK - BANK MONTREAL QUEBEC"
  Assets:Brokerage:BMO         -13 BMO {} @ 76.03 USD
    date: 2020-12-30
    ofx_fitid: "XXX"
    ofx_memo: "BANK MONTREAL QUEBEC"
    ofx_type: "SELLSTOCK"
  Income:Capital-gains:BMO
  Assets:Brokerage:Cash     988.37 USD
    ofx_fitid: "XXX"
  Expenses:Fees               0.02 USD

2020-12-30 open Assets:Brokerage:BMO                            BMO

2020-12-30 open Income:Capital-gains:BMO                        USD

2020-12-30 open Assets:Brokerage:Cash                           USD
```

If I edit that transaction so that it instead reads:

```
2020-12-30 * "SELLSTOCK - BANK MONTREAL QUEBEC"
  Assets:Brokerage:BMO         -13 BMO {} @ 76.03 USD
  Income:Capital-gains:BMO
  Assets:Brokerage:Cash     988.37 USD
  Expenses:Fees               0.02 USD
```

then suddenly `beancount-import` won't match it any more, and wants to add another copy of the transaction. I was not able to get into a situation where `beancount-import` is willing to augment a transaction I've already entered by adding in the appropriate metadata.

Here is my `main.beancount` with account ID censored:

```
2000-01-01 open Income:Capital-gains
2000-01-01 open Income:Dividends
2000-01-01 open Income:Interest
2000-01-01 open Expenses:Fees

1792-04-02 commodity USD
  cusip: "9999101"

2000-01-01 open Assets:Brokerage
  ofx_org: ""
  ofx_broker_id: "Wells Fargo Advisors"
  ofx_account_type: "securities_and_cash"
  account_id: "XXX"
  capital_gains_account: Income:Capital-gains
  fees_account: Expenses:Fees
  div_income_account: Income:Dividends
  interest_income_account: Income:Interest
```

`run_beancount_import.py` contains:

```
import beancount_import.webserver

def run_reconcile():
    data_sources = [
        {
            "module": "beancount_import.source.ofx",
            "ofx_filenames": ("export.ofx",)
        },
    ]

    beancount_import.webserver.main(
        argv = (),
        journal_input = "main.beancount",
        ignored_journal = "main.beancount",
        default_output = "main.beancount",
        open_account_output = "main.beancount",
        balance_account_output = "main.beancount",
        data_sources = data_sources,
    )

if __name__ == "__main__":
    run_reconcile()
```

If any details from `export.ofx` would be useful, let me know.

### PR DESCRIPTION
**Pull request by m-d-brown**
This is needed for 401(k) accounts, which can have an additional
sub-account like 'PreTax' to show the source of contributions.

### CODE REVIEW

File: beancount_import/source/ofx.py [hunk 1]
```diff
@@ -1336,6 +1328,22 @@ def _process_journal_entries(self):
                     InvalidSourceReference(excess_number, transactions))
 
 
+def _findOfxId(account, account_to_ofx_id):
```
Comment (jbms) [inline beancount_import/source/ofx.py:1331]: Can you change to `snake_case` for style consistency?

Comment (m-d-brown) [reply beancount_import/source/ofx.py]: Oops, had Go on my mind. Done. I've also added a second commit with a regression test that fails without the fix.

**Comment by m-d-brown** [comment]
I've also included d6c0cc4 to fix an issue with the paypal regression test.

**Comment by jbms** [comment]
Thanks, this looks good, the paypal test is now fixed so you can remove that change.

**Comment by m-d-brown** [comment]
Removed the paypal test change.

**Comment by jbms** [comment]
Thanks!

**Pull request merged**